### DISCUSSION TITLE
Unable to run Cypress in Docker on M1 Mac

### DISCUSSION THREAD

**Discussion by gerad** [cypress-io/cypress#15461] [Docker]
I know this is an edge case on an edge case (but hopefully one that's going to become more common). I'm having trouble running Cypress in Docker on an M1 Mac.

This is the error that I'm getting:

```
  cypress:cli Smoke test failed: Error: Command failed with exit code 255: /home/node/.cache/Cypress/6.6.0/Cypress/Cypress --no-sandbox --smoke-test --ping=367
qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory
    at makeError (/workspaces/site/node_modules/execa/lib/error.js:59:11)
    at handlePromise (/workspaces/site/node_modules/execa/index.js:114:26)
    at processTicksAndRejections (internal/process/task_queues.js:93:5) {
  shortMessage: 'Command failed with exit code 255: /home/node/.cache/Cypress/6.6.0/Cypress/Cypress --no-sandbox --smoke-test --ping=367',
  command: '/home/node/.cache/Cypress/6.6.0/Cypress/Cypress --no-sandbox --smoke-test --ping=367',
  exitCode: 255,
  signal: undefined,
  signalDescription: undefined,
  stdout: '',
  stderr: "qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory",
  failed: true,
  timedOut: false,
  isCanceled: false,
  killed: false
} +5ms
  cypress:cli error message: qemu-x86_64: Could not open '/lib64/ld-linux-x86-64.so.2': No such file or directory +1ms
  cypress:cli Stopping Xvfb +30ms
```

**Comment by gerad** [comment]
I can try to debug. But would appreciate some pointers about where to get started.

**Reply by bahmutov** [reply]
Hard to say, maybe read this first https://www.cypress.io/blog/2021/01/20/running-cypress-on-the-apple-m1-silicon-arm-architecture-using-rosetta-2/ but not sure about the Docker part there

**Reply by gerad** [reply]
Thank you. That is helpful. I think it won't be possible until you're cross compiling the binary for arm64. 

I can run the entire docker image under QEMU with x64. That'll slow things down, but I'm not sure how much – I can poke at that.

**Reply by Chekote** [reply]
Has there been any progress on this? That article was written almost 1.5yrs ago. Is there anywhere that I can track the progress of making Cypress compile under ARM?

**Reply by stokrattt** [reply]
I have to not see progress compiled under the arm. Cypress workly very slowly for this arm and it is bad.
